File: z-clients/python/diom/apis/msgs_sink.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    ListResponseSinkOut,
    SinkConfigureIn,
    SinkConfigureOut,
    SinkDeleteIn,
    SinkDeleteOut,
    SinkListIn,
)

from ..models.sink_list_in import _SinkListIn


class MsgsSinkAsync(ApiBase):
    async def configure(
        self,
        sink_configure_in: SinkConfigureIn,
    ) -> SinkConfigureOut:
        """Create or update a sink for a topic. Overwrites any existing sink with the same id."""
        body = sink_configure_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.sink.configure",
            body=body,
            response_type=SinkConfigureOut,
        )

    async def delete(
        self,
        sink_delete_in: SinkDeleteIn,
    ) -> SinkDeleteOut:
        """Delete a sink."""
        body = sink_delete_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.sink.delete",
            body=body,
            response_type=SinkDeleteOut,
        )

    async def list(
        self,
        topic: str,
        sink_list_in: SinkListIn = SinkListIn(),
    ) -> ListResponseSinkOut:
        """List sink configurations for a topic."""
        body = _SinkListIn(
            namespace=sink_list_in.namespace,
            topic=topic,
            limit=sink_list_in.limit,
            iterator=sink_list_in.iterator,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.sink.list",
            body=body,
            response_type=ListResponseSinkOut,
        )


class MsgsSink(ApiBase):
    def configure(
        self,
        sink_configure_in: SinkConfigureIn,
    ) -> SinkConfigureOut:
        """Create or update a sink for a topic. Overwrites any existing sink with the same id."""
        body = sink_configure_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.sink.configure",
            body=body,
            response_type=SinkConfigureOut,
        )

    def delete(
        self,
        sink_delete_in: SinkDeleteIn,
    ) -> SinkDeleteOut:
        """Delete a sink."""
        body = sink_delete_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.sink.delete",
            body=body,
            response_type=SinkDeleteOut,
        )

    def list(
        self,
        topic: str,
        sink_list_in: SinkListIn = SinkListIn(),
    ) -> ListResponseSinkOut:
        """List sink configurations for a topic."""
        body = _SinkListIn(
            namespace=sink_list_in.namespace,
            topic=topic,
            limit=sink_list_in.limit,
            iterator=sink_list_in.iterator,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.sink.list",
            body=body,
            response_type=ListResponseSinkOut,
        )

File: z-clients/rust/src/api/msgs_sink.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct MsgsSink<'a> {
    cfg: &'a Configuration,
}

impl<'a> MsgsSink<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Create or update a sink for a topic. Overwrites any existing sink with the same id.
    pub async fn configure(&self, sink_configure_in: SinkConfigureIn) -> Result<SinkConfigureOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.sink.configure")
            .with_body(sink_configure_in)
            .execute(self.cfg)
            .await
    }

    /// Delete a sink.
    pub async fn delete(&self, sink_delete_in: SinkDeleteIn) -> Result<SinkDeleteOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.sink.delete")
            .with_body(sink_delete_in)
            .execute(self.cfg)
            .await
    }

    /// List sink configurations for a topic.
    pub async fn list(
        &self,
        topic: String,
        sink_list_in: SinkListIn,
    ) -> Result<ListResponseSinkOut> {
        let sink_list_in = SinkListIn_ {
            namespace: sink_list_in.namespace,
            topic,
            limit: sink_list_in.limit,
            iterator: sink_list_in.iterator,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.sink.list")
            .with_body(sink_list_in)
            .execute(self.cfg)
            .await
    }
}
